### PR TITLE
Adjust to new Flint version 2.6.2

### PR DESCRIPTION
**Pull request by thofma**
As promised in https://github.com/oscar-system/LoadFlint.jl/issues/7#issuecomment-667516905 with the monkey patch to pick the right version at build time.

### CODE REVIEW

**Comment by wbhart** [comment]
What is revdeps full? And why are the tests associated with it failing? Is that safe to ignore?

**Comment by thofma** [comment]
They test Polymake.jl and Nemo.jl with the changes proposed here. They are allowed to fail, since Polymake.jl and Nemo.jl have not adjusted to the new version (which is merged here).

File: deps/build.jl [hunk 1]
```diff
@@ -40,9 +43,16 @@ elseif VERSION < v"1.3.0-rc4"
...
-     "build_FLINT.v2.6.0.jl",
    ]
 
+   if ver == v"2.6.0+0"
+     push!(dependencies, "build_FLINT.v2.6.0.jl")
+   elseif ver == v"2.6.2+0"
+     push!(dependencies, "build_FLINT.v2.6.2.jl")
+   else
+     throw(error("Flint version $ver not supported for julia version <= 1.3"))
+   end
```
Comment (fingolfin) [inline deps/build.jl:54]: Why exactly is this necessary? Can't we just change `Project.toml` to require FLINT_jll 2.6.2 or newer?

**Comment by thofma** [comment]
Before we agreed to put no upper bound or only lose upper bounds here. Probably something has changed that I am not aware off.

Also we would have to pin 2.6.2 exactly.

**Comment by thofma** [comment]
I think we agreed to let the packages downstream handle the compability with flint by putting combat bounds for FLINT_jll themselves.

**Comment by thofma** [comment]
See https://github.com/oscar-system/LoadFlint.jl/issues/7#issuecomment-651648360

**Comment by thofma** [comment]
I want to move forward with using the new flint version, so it would be good if we could agree on something.

@fingolfin @benlorenz

**Comment by thofma** [comment]
I will merge this soonish.

**Comment by fingolfin** [comment]
BTW can we remove build_FLINT.v0.0.1.jl and build_FLINT.v0.0.2.jl ?

**Comment by thofma** [comment]
Yes, good point. I will remove them and then merge the PR.

**Pull request merged**